src/Features/Core/Portable/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -259,25 +259,19 @@ public void CleanupGeneratedFiles()
             {
                 if (Directory.Exists(_rootTemporaryPath))
                 {
-                    var deletedEverything = true;
-
                     // Let's look through directories to delete.
                     foreach (var directoryInfo in new DirectoryInfo(_rootTemporaryPath).EnumerateDirectories())
                     {
-                        // Is there a mutex for this one?
+                        // Is there a mutex for this one?  If so, that means it's a folder open in another VS instance.
+                        // We should leave it alone.
                         if (Mutex.TryOpenExisting(CreateMutexName(directoryInfo.Name), out var acquiredMutex))
                         {
                             acquiredMutex.Dispose();
-                            deletedEverything = false;
-                            continue;
                         }
-
-                        TryDeleteFolderWhichContainsReadOnlyFiles(directoryInfo.FullName);
-                    }
-
-                    if (deletedEverything)
-                    {
-                        Directory.Delete(_rootTemporaryPath);
```

Comment:
no need to delete the topmost directory.  It's fine for there to be one top level temp folder we create once and which sticks around across sessions (note: we delete the contents within).  This just keeps thigns simpler.